File: tests/fixtures/php/safe.php
<?php
// SHOULD NOT auto-fix: side-effecting call being moved earlier would change behaviour
if (createUser() && sendEmail()) {
    echo "done";
}

// SHOULD NOT auto-fix: null guard — $obj->method() must stay after null check
if ($obj !== null && $obj->method()) {
    echo "ok";
}

// SHOULD NOT auto-fix: ignore comment
if ($expensive->call() && $cheap) { // condition-order: ignore
    echo "ignored";
}

File: tests/fixtures/go/safe.go
package main

func safeExamples(obj *MyStruct, err error) {
	// SHOULD NOT auto-fix: nil guard dependency
	if obj != nil && obj.IsActive() {
		println("ok")
	}

	// SHOULD NOT auto-fix: error check must stay first
	if err != nil && err.Error() == "timeout" {
		println("err")
	}
}

type MyStruct struct{ active bool }

func (m *MyStruct) IsActive() bool { return m.active }
